internal/syncd/integrations/ecr/reconciler_test.go

```diff
@@ -107,3 +108,30 @@ func TestReconcilerSuccess(t *testing.T) {
 	mockIndexer.AssertExpectations(t)
 	mockEditor.AssertExpectations(t)
 }
+
+func TestReconcilerNoRegisteredReleaseAffected(t *testing.T) {
+	mockEditor := new(MockReleaseEditor)
+	mockIndexer := new(MockReleaseIndexer)
+
+	reconciler := NewReconciler(mockEditor, mockIndexer)
+
+	inputImage := &image.Ref{
+		Registry:   "723255503624.dkr.ecr.us-east-1.amazonaws.com",
+		Repository: "bar",
+		Tag:        "78bc9ccf64eb838c6a0e0492ded722274925e2bd",
+	}
+
+	releaseNames := []types.NamespacedName{
+		{
+			Namespace: "default",
+			Name:      "bar",
+		},
+	}
+
+	mockIndexer.On("Lookup", inputImage).Return(releaseNames, error(nil))
+
+	mockEditor.On("Edit", mock.Anything, releaseNames, inputImage).Return(errors.Wrap(errNoRegisteredReleasesAffected, "extra context"))
+
+	err := reconciler.Reconcile(context.Background(), inputImage)
+	assert.NoError(t, err)
```

Comment:
The expectation is that this is an error correct?

Reply:
When the editor returns `errNoRegisteredReleasesAffected`, the reconciler should log the error but return `nil` because it successfully reconciled 0 releases. So the `NoError` assertion is correct here.